README.md

```diff
@@ -1,3 +1,36 @@
 # scalaz-schema
 
-[![Gitter](https://badges.gitter.im/scalaz/scalaz-schema.svg)](https://gitter.im/scalaz/scalaz-schema?utm_source=badge&utm_medium=badge&utm_campaign=pr-badge&utm_content=badge)
\ No newline at end of file
+[![Gitter](https://badges.gitter.im/scalaz/scalaz-schema.svg)](https://gitter.im/scalaz/scalaz-schema?utm_source=badge&utm_medium=badge&utm_campaign=pr-badge&utm_content=badge)
+
+
+## Goal
+
+A purely-functional library for defining type-safe schemas for algebraic data types, providing free generators, SQL queries, JSON codecs, binary codecs, and migration from this schema definition.
+
+## Introduction & Highlights
+
+Scalaz Schema defines a generic representation of algebraic data structures and combinators that turn such schema into a generic computation over arbitrary data. In other words, Scalaz Schema provides a way to express any computation that abstracts over the structure of data, such as:
+
+* Codecs: given a serial format (binary, JSON, etc.) and the schema of a data structure (say, an ADT) we can derive a codec for that data structure and serial format.
+* Data Generators: given any schema we an derive random data generators (eg. scalacheck's `Gen`) that produce data satisfying that schema.
+* Schema/Data Migrations: since schemas are values, we can easily verify wether two versions of a schema are forward/backward compatible and provide a generic way to upgrade/downgrade data from one version of the schema to the other.
+* Diffing/Patching: given a schema we can generically compute the difference between two pieces of data satisfying that schema. In the same spirit, we have generic way to apply patches to apply patches to arbitrary data structures.
```

Comment:
typo: `to apply patches to apply patches`

Reply:
"copy that" if I may say ^^